### PR TITLE
Feature 1661 - hzn exchange service list to truncate the clusterDeplo…

### PR DESCRIPTION
**Pull request by linggao**
…yment string from output

### CODE REVIEW

File: cli/hzn.go [hunk 1]
```diff
@@ -235,7 +235,8 @@ Environment Variables:
...
 	exServiceListNodeIdTok := exServiceListCmd.Flag("node-id-tok", msgPrinter.Sprintf("The Horizon Exchange node ID and token to be used as credentials to query and modify the node resources if -u flag is not specified. HZN_EXCHANGE_NODE_AUTH will be used as a default for -n. If you don't prepend it with the node's org, it will automatically be prepended with the -o value.")).Short('n').PlaceHolder("ID:TOK").String()
-	exServiceLong := exServiceListCmd.Flag("long", msgPrinter.Sprintf("When listing all of the services, show the entire resource of each services, instead of just the name.")).Short('l').Bool()
+	exServiceLong := exServiceListCmd.Flag("long", msgPrinter.Sprintf("When listing all of the services, show the entire resource of each services, instead of just the name. When listing a specific service, show more details.")).Short('l').Bool()
+	exSvcOpYamlFilePath := exServiceListCmd.Flag("op-yaml-file", msgPrinter.Sprintf("The name of the file where the Kubernetes operator yaml archive will be saved. The saving will happen only when one service is specified and the clusterDeployment attribute is not empty for the service.")).Short('f').String()
```
Comment (dabooz) [inline cli/hzn.go:239]: Change "The saving will happen only when one service is specified and the clusterDeployment attribute is not empty for the service." to "This flag is only used when listing a specific service. This flag is ignored when the service does not have a clusterDeployment attribute."

File: cli/hzn.go [hunk 2]
```diff
@@ -235,7 +235,8 @@ Environment Variables:
...
 	exService := exServiceListCmd.Arg("service", msgPrinter.Sprintf("List just this one service. Use <org>/<svc> to specify a public service in another org, or <org>/ to list all of the public services in another org.")).String()
 	exServiceListNodeIdTok := exServiceListCmd.Flag("node-id-tok", msgPrinter.Sprintf("The Horizon Exchange node ID and token to be used as credentials to query and modify the node resources if -u flag is not specified. HZN_EXCHANGE_NODE_AUTH will be used as a default for -n. If you don't prepend it with the node's org, it will automatically be prepended with the -o value.")).Short('n').PlaceHolder("ID:TOK").String()
-	exServiceLong := exServiceListCmd.Flag("long", msgPrinter.Sprintf("When listing all of the services, show the entire resource of each services, instead of just the name.")).Short('l').Bool()
+	exServiceLong := exServiceListCmd.Flag("long", msgPrinter.Sprintf("When listing all of the services, show the entire resource of each services, instead of just the name. When listing a specific service, show more details.")).Short('l').Bool()
```
Comment (dabooz) [inline cli/hzn.go:238]: change "show the entire resource of each services," to "show the entire service definition,"

File: cli/exchange/service.go [hunk 3]
```diff
@@ -67,11 +74,44 @@ func ServiceList(credOrg, userPw, service string, namesOnly bool) {
...
+		jsonBytes, err := json.MarshalIndent(exchServices, "", cliutils.JSON_INDENT)
 		if err != nil {
 			cliutils.Fatal(cliutils.JSON_PARSING_ERROR, msgPrinter.Sprintf("failed to marshal 'hzn exchange service list' output: %v", err))
 		}
 		fmt.Println(string(jsonBytes))
+
+		// save the kube operator yaml archive to file if filePath is specified and one service is specified
+		if filePath != "" {
+			if clusterDeployment == "" {
+				msgPrinter.Printf("Skip file saving because the clusterDepolyment attribute is empty for this service.")
```
Comment (dabooz) [inline cli/exchange/service.go:109]: change "Skip file saving because" to "Ignoring -f because"

File: cli/exchange/service.go [hunk 4]
```diff
@@ -563,3 +603,58 @@ func ServiceRemovePolicy(org string, credToUse string, service string, force boo
...
+	archiveData := GetOpYamlArchiveFromClusterDepl(clusterDeployment)
+
+	fileName := filePath
+	if info, err := os.Stat(filePath); err == nil && info.IsDir() {
+		_, id := cliutils.TrimOrg("", sId) // remove the org part
+		fileName = filepath.Join(filePath, id+"_operator_yaml.tar.gz")
+	}
+
+	if _, err := os.Stat(fileName); err == nil {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("File %v already exist. Please specify a different file name.", fileName))
```
Comment (dabooz) [inline cli/exchange/service.go:621]: change "already exist" to "already exists"

File: cli/exchange/service.go [hunk 5]
```diff
@@ -563,3 +603,58 @@ func ServiceRemovePolicy(org string, credToUse string, service string, force boo
...
+
+	archiveData := GetOpYamlArchiveFromClusterDepl(clusterDeployment)
+
+	fileName := filePath
+	if info, err := os.Stat(filePath); err == nil && info.IsDir() {
+		_, id := cliutils.TrimOrg("", sId) // remove the org part
+		fileName = filepath.Join(filePath, id+"_operator_yaml.tar.gz")
+	}
+
+	if _, err := os.Stat(fileName); err == nil {
```
Comment (dabooz) [inline cli/exchange/service.go:620]: If the file already exists we should ask the user if they want us to overwrite it anyway....maybe also support the -F option, which means force overwrite and we dont have to ask.

File: cli/exchange/service.go [hunk 6]
```diff
@@ -563,3 +603,58 @@ func ServiceRemovePolicy(org string, credToUse string, service string, force boo
...
+	if err != nil {
+		cliutils.Fatal(cliutils.INTERNAL_ERROR, msgPrinter.Sprintf("Failed to create file %v. %v", fileName, err))
+	}
+	defer file.Close()
+
+	if _, err := file.Write(archiveData); err != nil {
+		cliutils.Fatal(cliutils.INTERNAL_ERROR, msgPrinter.Sprintf("Failed to save the Kube operator yaml archive data to file %v.", fileName))
+	}
+
+	msgPrinter.Printf("The Kube operator yaml archive is saved to file %v", fileName)
```
Comment (dabooz) [inline cli/exchange/service.go:634]: Change "The Kube operator yaml" to "The clusterDeployment operator yaml"

File: cli/exchange/service.go [hunk 7]
```diff
@@ -45,6 +47,11 @@ func ServiceList(credOrg, userPw, service string, namesOnly bool) {
 	if service == "*" {
 		service = ""
 	}
+
+	if service == "" && filePath != "" {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("-f can only be used when one servie is specified."))
```
[the remark this answers is not in the document]
Comment (linggao) [reply cli/exchange/service.go:52]: fixed. thanks.

File: cli/hzn.go (hunk 1 above)
Comment (linggao) [reply cli/hzn.go:239]: done

File: cli/hzn.go (hunk 2 above)
Comment (linggao) [reply cli/hzn.go:238]: done

File: cli/exchange/service.go (hunk 4 above)
Comment (linggao) [reply cli/exchange/service.go:621]: done

File: cli/exchange/service.go (hunk 3 above)
Comment (linggao) [reply cli/exchange/service.go:109]: done

File: cli/exchange/service.go (hunk 6 above)
Comment (linggao) [reply cli/exchange/service.go:634]: done

File: cli/exchange/service.go (hunk 5 above)
Comment (linggao) [reply cli/exchange/service.go:620]: done. added -F.

**Pull request merged**